src/test/java/edu/ucla/library/manifeststore/ManifestUploadIT.java

```diff
@@ -0,0 +1,194 @@
+                });
+                asyncTask.complete();
+            } else {
+                aContext.fail(LOGGER.getMessage(MessageCodes.MFS_031));
+            }
+
+        }); // end HttpClient.getNow
+    } // end method
+
+    /**
+     * confirm that it's possible to PUT a manifest
+     */
+    @SuppressWarnings("deprecation")
+    @Test
+    public final void TestBcheckThatPUTmanifestWorks(final TestContext aContext) {
+        final Async asyncTask = aContext.async();
+
+        /************ PUT TEST ************************************************/
+        final String myDotJsonPutManifestID = Constants.PUT_TEST_ID_PREFIX + MANIFEST_ID_WITH_EXT;
+        final String testIDPath = StringUtils.format(MANIFEST_PATH, myDotJsonPutManifestID);
+        LOGGER.info(MessageCodes.MFS_016, testIDPath); // Test PUTing a test manifest to: {}
+        final Buffer myManifest = vertx.fileSystem().readFileBlocking(myManifestFilePath);
+        // set up our requestOpts for this PUT
+        final RequestOptions requestOpts = new RequestOptions();
+        requestOpts.setPort(PORT).setHost(Constants.UNSPECIFIED_HOST).setURI(testIDPath);
+        requestOpts.addHeader(Constants.CONTENT_TYPE, Constants.JSON_MEDIA_TYPE);
+
+        // PUT this Manifest
+        vertx.createHttpClient().put(requestOpts, response -> {
+            final int statusCode = response.statusCode();
+            if (statusCode == HTTP.OK) {
+                asyncTask.complete();
+            } else {
+                aContext.fail(MessageCodes.MFS_032);
+            }
+        }).end(myManifest); // end HttpClient.put
+    } // end method
+
+    /**
+     * confirm that it's possible to GET a manifest
+     *
+     * @throws IOException
+     */
+    @SuppressWarnings("deprecation")
+    @Test
+    public final void TestCcheckThatGETmanifestWorks(final TestContext aContext) throws IOException {
+        final Async asyncTask = aContext.async();
+        /************ GET TEST ************************************************/
+        final String myDotJsonPutManifestID = Constants.PUT_TEST_ID_PREFIX + MANIFEST_ID_WITH_EXT;
+        final String testIDPath = StringUtils.format(MANIFEST_PATH, myDotJsonPutManifestID);
+        final String expectedManifest = StringUtils.read(MANIFEST_FILE);
+
+        LOGGER.info(MessageCodes.MFS_027, testIDPath); // Test GETing a test manifest from: {}
+        vertx.createHttpClient().getNow(PORT, Constants.UNSPECIFIED_HOST, testIDPath, response -> {
+            final int statusCode = response.statusCode();
+            if (statusCode == HTTP.OK) {
+                // yep, we can GET a manifest, one last thing to check
+                response.bodyHandler(body -> {
+                    final String foundManifest = body.toString(StandardCharsets.UTF_8);
+                    // Check that what we retrieve is the same as what we stored
+                    if (new JsonObject(expectedManifest).equals(new JsonObject(foundManifest))) {
```

Comment:
You should put the asyncTask.complete() in the `if` block since the else block completes the context by failing the test.